src/test/java/com/coveo/pushapiclient/DocumentUploadQueueTest.java

```diff
@@ -0,0 +1,192 @@
+package com.coveo.pushapiclient;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class DocumentUploadQueueTest {
+
+    @Mock
+    private UploadStrategy uploadStrategy;
+
+    @InjectMocks
+    private DocumentUploadQueue queue;
+
+    private AutoCloseable closeable;
+    private DocumentBuilder documentToAdd;
+    private DeleteDocument documentToDelete;
+
+    private int oneMegaByte = 1 * 1024 * 1024;
+
+    private String generateStringFromBytes(int numBytes) {
+        // Check if the number of bytes is valid
+        if (numBytes <= 0) {
+            return "";
+        }
+
+        // Create a byte array with the specified length
+        byte[] bytes = new byte[numBytes];
+
+        // Fill the byte array with a pattern of ASCII characters
+        byte pattern = 65; // ASCII value for 'A'
+        for (int i = 0; i < numBytes; i++) {
+            bytes[i] = pattern;
+        }
+
+        return new String(bytes);
+    }
+
+    private DocumentBuilder generateDocumentFromSize(int numBytes) {
+        return new DocumentBuilder("https://my.document.uri?ref=1",
+                "My bulky document")
+                .withData(generateStringFromBytes(numBytes));
+    }
+
+    @Before
+    public void setup() {
+        String twoMegaByteData = generateStringFromBytes(2 * oneMegaByte);
+
+        documentToAdd = new DocumentBuilder(
+                "https://my.document.uri?ref=1",
+                "My new document")
+                .withData(twoMegaByteData);
+
+        documentToDelete = new DeleteDocument("https://my.document.uri?ref=3");
+
+        closeable = MockitoAnnotations.openMocks(this);
+    }
+
+    @After
+    public void closeService() throws Exception {
+        closeable.close();
+    }
+
+    @Test
+    public void testIsEmpty() throws IOException, InterruptedException {
+        assertTrue(queue.isEmpty());
+    }
+
+    @Test
+    public void testIsNotEmpty() throws IOException, InterruptedException {
+        queue.add(documentToAdd);
```

Comment:
would be better if the queue is cleared before each test

Reply:
No need. Mockito already takes care of that.
A new queue instance is generated for each test.
Mockito will to instantiate the `DocumentUploadQueue` object with the appropriate mocks `@Before` each test. 
```java
@InjectMocks
private DocumentUploadQueue queue;
```
The `MockitoAnnotations.openMocks()` method returns an instance of `AutoClosable` which can be used to close the resource after the test.
```java
closeable = MockitoAnnotations.openMocks(this);
// ...
@After
public void closeService() throws Exception {
    closeable.close();
}
```